devito/ir/cgen/printer.py

```diff
@@ -370,7 +370,7 @@ def _print_UnaryOp(self, expr, op=None, parenthesize=False):
 
     def _print_Cast(self, expr):
         cast = f'({self._print(expr._C_ctype)}{self._print(expr.stars)})'
-        return self._print_UnaryOp(expr, op=cast)
+        return self._print_UnaryOp(expr, op=cast, parenthesize=not expr.base.is_Atom)
```

Comment:
I don't think this is safe actually? you could have `Cast(a*b)` and `a*b` has no base, so it would crash with an AttributeError. Why not just `expr.is_Atom`?

Reply:
expr is a `Cast` which has
-  `Cast.base` , the actual expression
- `Cast.stars` stars for pointer
- `Cast. _C_ctype` the type to cast to

`expr.is_Atom` ill always be False be cause that would check if the Cast object is an Atom not its expression

Reply:
And `expr.base` is always a sympy object by construction so .is_Atom will not crash

https://github.com/devitocodes/devito/blob/598fd62ec11b0df54b42f818e33aa8f5e923f89a/devito/symbolics/extended_sympy.py#L336

Reply:
oooohh right, I had forgotten about its superclass !